components/properties/src/runtime.rs

```diff
@@ -357,4 +357,229 @@ impl UnicodeProperty {
 
         Some(prop)
     }
+
+    /// Returns a type capable of looking up values for a property specified as a string, as long as it is a
+    /// [binary property listed in ECMA-262][ecma], using strict matching on the names in the spec.
+    ///
+    /// This handles every property required by ECMA-262 `/u` regular expressions, except for:
+    ///
+    /// - `Script` and `General_Category`: handle these directly with [`maps::load_general_category()`] and
+    ///    [`maps::load_script()`].
+    ///    using property values parsed via [`GeneralCategory::get_name_to_enum_mapper()`] and [`Script::get_name_to_enum_mapper()`]
+    ///    if necessary.
+    /// - `Script_Extensions`: handle this directly using APIs from [`crate::script`], like [`script::load_script_with_extensions_unstable()`]
+    /// - `General_Category` mask values: Handle this alongside `General_Category` using [`GeneralCategoryGroup`],
+    ///    using property values parsed via [`GeneralCategoryGroup::get_name_to_enum_mapper()`] if necessary
+    /// - `Assigned`, `All`, and `ASCII` pseudoproperties: Handle these using their equivalent sets:
+    ///    - `Any` can be expressed as the range `[\u{0}-\u{10FFFF}]`
+    ///    - `Assigned` can be expressed as the inverse of the set `gc=Cn` (i.e., `\P{gc=Cn}`).
+    ///    - `ASCII` can be expressed as the range `[\u{0}-\u{7F}]`
+    /// - `General_Category` property values can themselves be treated like properties using a shorthand in ECMA262,
+    ///    simply create the corresponding `GeneralCategory` set.
+    ///
+    /// ✨ *Enabled with the `compiled_data` Cargo feature.*
+    ///
+    /// [📚 Help choosing a constructor](icu_provider::constructors)
+    ///
+    /// ```
+    /// use icu::properties::runtime::UnicodeProperty;
+    ///
+    /// let emoji = UnicodeProperty::parse_ecma262_name("Emoji").unwrap().load().expect("loading data failed");
+    ///
+    /// assert!(emoji.contains('🔥')); // U+1F525 FIRE
+    /// assert!(!emoji.contains('V'));
+    /// ```
+    ///
+    /// [ecma]: https://tc39.es/ecma262/#table-binary-unicode-properties
+    #[cfg(feature = "compiled_data")]
+    pub fn load(&self) -> Result<crate::sets::CodePointSetDataBorrowed<'static>, DataError> {
```

Comment:
issue: this removes the ECMA262 naming from the methods, why? UnicodeProperty supports more properties than ECMA262, and we wanted to clarify that this is only loading that subset.

Reply:
> UnicodeProperty supports more properties than ECMA262

Didn't know, that, this is probably not the enum we need then.

Reply:
I want an intermediate step for the two error cases in `load_for_ecma262_unstable`, parsing the str and loading the data. I thought `UnicodeProperty` was that intermediate step.